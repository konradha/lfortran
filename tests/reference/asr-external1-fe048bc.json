{
    "basename": "asr-external1-fe048bc",
    "cmd": "lfortran --indent --show-asr --implicit-typing --no-color {infile} -o {outfile}",
    "infile": "tests/../integration_tests/external1.f90",
    "infile_hash": "d1e5f47338a407a9c31e7dc02f301593a20de1be72b878859b097486",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-external1-fe048bc.stdout",
    "stdout_hash": "f168f0c9f201c1777d5906e4b88bf3b37376c13d69d01270cacba904",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}