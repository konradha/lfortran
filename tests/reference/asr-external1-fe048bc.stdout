(TranslationUnit 
    (SymbolTable
        1
        {
            fun1:
                (Function 
                    (SymbolTable
                        4
                        {
                            a:
                                (Variable 
                                    4 
                                    a 
                                    Local 
                                    () 
                                    () 
                                    Default 
                                    (Real 4 []) 
                                    Source 
                                    Public 
                                    Required 
                                    .false.
                                )
                            
                        }) 
                    fun1 
                    [] 
                    [] 
                    [(= 
                        (Var 4 a) 
                        (Cast 
                            (IntegerConstant 100 (Integer 4 [])) 
                            IntegerToReal 
                            (Real 4 []) 
                            (RealConstant 
                                100.000000 
                                (Real 4 [])
                            )
                        ) 
                        ()
                    )
                    (= 
                        (Var 4 a) 
                        (RealBinOp 
                            (Var 4 a) 
                            Add 
                            (Cast 
                                (IntegerConstant 1 (Integer 4 [])) 
                                IntegerToReal 
                                (Real 4 []) 
                                (RealConstant 
                                    1.000000 
                                    (Real 4 [])
                                )
                            ) 
                            (Real 4 []) 
                            ()
                        ) 
                        ()
                    )] 
                    () 
                    Source 
                    Public 
                    Interface 
                    () 
                    .false. 
                    .false. 
                    .false. 
                    .false.
                ), 
            fun2:
                (Function 
                    (SymbolTable
                        5
                        {
                            
                        }) 
                    fun2 
                    [] 
                    [] 
                    [] 
                    () 
                    Source 
                    Public 
                    Interface 
                    () 
                    .false. 
                    .false. 
                    .false. 
                    .false.
                ), 
            g:
                (Function 
                    (SymbolTable
                        6
                        {
                            fun:
                                (Variable 
                                    6 
                                    fun 
                                    Unspecified 
                                    () 
                                    () 
                                    Default 
                                    (Integer 4 []) 
                                    Source 
                                    Public 
                                    Required 
                                    .false.
                                ), 
                            i:
                                (Variable 
                                    6 
                                    i 
                                    Local 
                                    () 
                                    () 
                                    Default 
                                    (Integer 4 []) 
                                    Source 
                                    Public 
                                    Required 
                                    .false.
                                ), 
                            proc:
                                (Variable 
                                    6 
                                    proc 
                                    Unspecified 
                                    () 
                                    () 
                                    Default 
                                    (Real 4 []) 
                                    Source 
                                    Public 
                                    Required 
                                    .false.
                                )
                            
                        }) 
                    g 
                    [(Var 6 proc)
                    (Var 6 fun)] 
                    [] 
                    [(= 
                        (Var 6 i) 
                        (ArrayItem 
                            (Var 6 fun) 
                            [] 
                            (Integer 4 []) 
                            ()
                        ) 
                        ()
                    )
                    (SubroutineCall 
                        6 proc 
                        () 
                        [((Var 6 i))] 
                        ()
                    )] 
                    () 
                    Source 
                    Public 
                    Implementation 
                    () 
                    .false. 
                    .false. 
                    .false. 
                    .false.
                ), 
            main:
                (Program 
                    (SymbolTable
                        2
                        {
                            fun2:
                                (Variable 
                                    2 
                                    fun2 
                                    Local 
                                    () 
                                    () 
                                    Default 
                                    (Integer 4 []) 
                                    Source 
                                    Public 
                                    Required 
                                    .false.
                                )
                            
                        }) 
                    main 
                    [] 
                    [(SubroutineCall 
                        1 g 
                        () 
                        [((Var 1 sub1))
                        ((Var 1 fun1))] 
                        ()
                    )]
                ), 
            sub1:
                (Function 
                    (SymbolTable
                        3
                        {
                            fun1:
                                (Function 
                                    (SymbolTable
                                        7
                                        {
                                            a:
                                                (Variable 
                                                    7 
                                                    a 
                                                    Local 
                                                    () 
                                                    () 
                                                    Default 
                                                    (Integer 4 []) 
                                                    Source 
                                                    Public 
                                                    Required 
                                                    .false.
                                                ), 
                                            fun1:
                                                (Variable 
                                                    7 
                                                    fun1 
                                                    ReturnVar 
                                                    () 
                                                    () 
                                                    Default 
                                                    (Integer 4 []) 
                                                    Source 
                                                    Public 
                                                    Required 
                                                    .false.
                                                )
                                            
                                        }) 
                                    fun1 
                                    [] 
                                    [] 
                                    [] 
                                    (Var 7 fun1) 
                                    Source 
                                    Public 
                                    Implementation 
                                    () 
                                    .false. 
                                    .false. 
                                    .false. 
                                    .false.
                                ), 
                            i:
                                (Variable 
                                    3 
                                    i 
                                    Unspecified 
                                    () 
                                    () 
                                    Default 
                                    (Integer 4 []) 
                                    Source 
                                    Public 
                                    Required 
                                    .false.
                                )
                            
                        }) 
                    sub1 
                    [(Var 3 i)] 
                    [] 
                    [(FileWrite 
                        0 
                        () 
                        () 
                        () 
                        () 
                        () 
                        [(Var 3 i)] 
                        () 
                        ()
                    )] 
                    () 
                    Source 
                    Public 
                    Interface 
                    () 
                    .false. 
                    .false. 
                    .false. 
                    .false.
                )
            
        }) 
    []
)
